src/classes/CompensationController.cls

```diff
@@ -0,0 +1,35 @@
+public with sharing class CompensationController {
+	@AuraEnabled
+    public static List<Compensation__c> getCompensations(String filter){
+		List<Compensation__c> compensations = new List<Compensation__c>();
+		if (filter == 'all'){
+			compensations = [SELECT Employee__r.Id, Employee__r.Name, Employee__r.birthdate,
```

Comment:
Please use dynamic SOQL instead of static conditional queries. That way you can form a single query that's modified dynamically based on the different conditions you have